modules/lang-painless/src/main/java/org/elasticsearch/painless/GenericElasticsearchScript.java

```diff
@@ -26,10 +26,14 @@
 /**
  * Generic script interface that Painless implements for all Elasticsearch scripts.
  */
-public interface GenericElasticsearchScript {
-    String[] ARGUMENTS = new String[] {"params", "_score", "doc", "_value", "ctx"};
-    Object execute(Map<String, Object> params, double _score, Map<String, ScriptDocValues<?>> doc, Object _value, Map<?, ?> ctx);
+public abstract class GenericElasticsearchScript {
 
-    boolean uses$_score();
-    boolean uses$ctx();
+    public GenericElasticsearchScript() {}
+
+    public static final String[] PARAMETERS = new String[] {"params", "_score", "doc", "_value", "ctx"};
```

Comment:
Given that we have a common parameter called params, can we call this `PARAMETER_NAMES`?